### PR TITLE
Deprecate mixing the `url` option with `host`, `dbname` and a few others

### PR DESCRIPTION
**Pull request by nicolas-grekas**
As proposed in #1331

For discussion. (I'll take care of tests once the discussion has settled.)

### CODE REVIEW

**Comment by ostrolucky** [comment]
You mean #1331, not #1334, right?

@stof I'm not quite sure what you meant with https://github.com/doctrine/DoctrineBundle/pull/1331#issuecomment-824912640, is that still a problem in this PR?

I like this PR though, IMO it's ok to finish it.

**Comment by nicolas-grekas** [comment]
I extracted the part about `override_url` in #1342 because deprecating the other options requires updating a lot of tests, and also because the discussion should be split apart.

**Comment by stof** [comment]
@ostrolucky this PR does not deprecate options like `server_version`, only the ones corresponding to the standard parts of the URL (the query string being used by DBAL for extra options is something totally specific to DBAL, relying on DBAL option names).
So it can be fine with me.

However, we should be careful when writing the upgrade guide (asking people to dynamically create a URL in the config file from separate username, host, password and db name is quite fragile, as it would probably miss the need for URL escaping, which will only trigger a bug after some time when rotating the password to one using special URL chars for instance)

**Comment by ostrolucky** [comment]
Just a reminder that after https://github.com/doctrine/DoctrineBundle/pull/1331 has been merged, we expect this to be finished next.

**Comment by nicolas-grekas** [comment]
I went with a more conservative approach that should provide the same benefit:

what is now deprecated is mixing the `url` option with `host`, `dbname` and a few others.

**Comment by ostrolucky** [comment]
Yep that's good compromise I think

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -245,12 +246,17 @@ private function configureDbalDriverNode(ArrayNodeDefinition $node): void
         $node
             ->validate()
             ->always(static function (array $values) {
-                $deprecatedOptions = ['override_url' => true];
-                $deprecatedValues  = array_intersect_key($values, $deprecatedOptions);
+                if (! isset($values['url'])) {
+                    return $values;
+                }
+
+                $urlConflictingOptions = ['host' => true, 'port' => true, 'user' => true, 'password' => true, 'path' => true, 'dbname' => true, 'unix_socket' => true, 'memory' => true];
```
Comment (ostrolucky) [inline DependencyInjection/Configuration.php:253]: I think driver is missing, isn't it?

Comment (nicolas-grekas) [reply DependencyInjection/Configuration.php:253]: `driver` is in another branch of the config tree.
I don't think that's an issue.

**Review by ostrolucky — APPROVED** [review]

**Comment by ostrolucky** [comment]
thx @nicolas-grekas

**Pull request merged**